src/fspec-fromdir.c

```diff
@@ -0,0 +1,113 @@
+#define _XOPEN_SOURCE 500
+#include <ftw.h>
+#include <limits.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+#include <errno.h>
+
+int root_owned = 0;
+int absolute = 0;
+char *prefix = "";
+
+static void
+check(int ok) {
```

Comment:
@michaelforney Part of me thinks this check definition is just lazy programming, though in practice it works well. Would you prefer I inlined error checks?

Reply:
It works well enough, but it is not perfect. It doesn't give much context on what operation failed, and although you set `errno = 0` at the top of the file, most POSIX functions can succeed and change `errno`, so the error message might not be correct.

I don't mind this for now,  but my current preferred approach is explicit error checking, with a fatal function like this:

```c
static void
fatal(const char *fmt, ...)
{
	va_list ap;

	va_start(ap, fmt);
	vfprintf(stderr, ap, fmt);
	va_end(ap);
	if (fmt[0] && fmt[strlen(fmt) - 1] == ':') {
		fputc(' ', stderr);
		perror(NULL);
	} else {
		fputc('\n', stderr);
	}
	exit(1);
}
```

Alternatively, you could use err.h, which is available on pretty much everywhere.